smoke_tests/smoke_tests.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+import sys
+import unittest
+from unittest import TestSuite
+import requests
+
+from tests.smoke_test_case import SmokeTestCase
+from tests.unauthenticated.status_tests import StatusTests
+from tests.authenticated.azure.managed_apps_tests import ManagedAppsTests
+from tests.authenticated.billing_profiles import BillingProfileTests
+
+DESCRIPTION = """
+BPM Smoke Test
+Enter the host (domain and optional port) of the BPM instance you want to to test. This test will ensure that the BPM 
+instance running on that host is minimally functional.
+"""
+
+
+def gather_tests(is_authenticated: bool = False, azure_subscription: bool = False) -> TestSuite:
+    suite = unittest.TestSuite()
+    status_tests = unittest.defaultTestLoader.loadTestsFromTestCase(StatusTests)
+    suite.addTests(status_tests)
+    if is_authenticated:
+        profile_tests = unittest.defaultTestLoader.loadTestsFromTestCase(BillingProfileTests)
+        suite.addTests(profile_tests)
+        if azure_subscription:
+            managed_apps_tests = unittest.defaultTestLoader.loadTestsFromTestCase(ManagedAppsTests)
+            suite.addTests(managed_apps_tests)
+    return suite
+
+
+def main(main_args):
+    if main_args.user_token:
+        verify_user_token(main_args.user_token)
+
+    SmokeTestCase.BPM_HOST = main_args.bpm_host
+    SmokeTestCase.USER_TOKEN = main_args.user_token
+    ManagedAppsTests.AZURE_SUBSCRIPTION_ID = main_args.azure_sub_id
+
+    test_suite = gather_tests(main_args.user_token, main_args.azure_sub_id)
+
+    runner = unittest.TextTestRunner(verbosity=main_args.verbosity)
+    runner.run(test_suite)
+
+
+def verify_user_token(user_token: str):
+    response = requests.get(f"https://www.googleapis.com/oauth2/v1/tokeninfo?access_token={user_token}")
+    assert response.status_code == 200, "User Token is no longer valid.  Please generate a new token and try again."
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        prog="BPM Smoke Tests",
+        description=DESCRIPTION,
+    )
+    parser.add_argument(
+        "bpm_host",
+        help="Required domain with optional port number of the BPM host you want to test"
+    )
+    parser.add_argument(
+        "user_token",
+        nargs='?',
+        default=None,
+        help="Optional. If present, will test additional authenticated endpoints using the specified token"
+    )
+    parser.add_argument(
+        "--azure_sub_id",
+        type=str,
+        help="Optional Azure SubscriptionId. If present, will test retrieving azure managed apps using the specified id"
```

Comment:
default value since it is optional? For `user_token` you specified a default of None.

Reply:
Why is the name `--azure_sub_id`, whereas `user_token` didn't have the `--` prefix?

Reply:
`user_token` is positional, where `--azure_sub_id` is not. 
I wanted to limit the number of of positional arguments generally, since too many of them, and more than one positional argument can get confusing quickly.
`user_token` is related to a much broader class of functionality: without it you can't test _any_ authenticated API, where azure_sub_id is a more specific functionality, so I felt it made sense for that to be positional/

Reply:
That makes sense, thanks.

Reply:
This is also why `--azure_sub_id` doesn't have default value - only positional arguments need to specify them.